Remove dependency on .envrc and update contribution

winkingturtle-vmw opened the pull request:
`.envc` has always existed and been relied upon by developers mostly for setting `GOPATH`. Since we've stopped supporting `GOPATH` and this dependency has not been used as actively, let's remove dependency on .envrc. Also re-organize scripts so that it's clear which scripts are ran inside of docker containers and which are ran in the local workstation. Updated README to reflect those changes.

- `./scripts/test-in-docker-locally.bash` -> Helper to run Everything
- `./scripts/create-docker-container.bash` -> Create a docker container to test and run things manually

Closes: cloudfoundry/routing-release#352
CI PR: https://github.com/cloudfoundry/wg-app-platform-runtime-ci/pull/13

# What type of change is this?

- [ ] `[Breaking Change]`: the change removes a feature or introduces a behavior change to core functionality (request routing, request logging)
- [ ] `[Minor Feature/Improvement]`:  the change introduces a new feature or improvement that doesn't alter core behavior
- [x] `[Bug Fix]`: the change addresses a defect


# Backwards Compatibility

Yes


# How should this be tested?

N.A

# Additional Context

https://github.com/cloudfoundry/routing-release/issues/352

# PR Checklist
* [x] I have viewed, signed, and submitted the Contributor License Agreement.
* [x] I have made this pull request to the `develop` branch.
* [x] I have [run all the unit tests](https://github.com/cloudfoundry/routing-release#running-unit-and-integration-tests).
* [ ] I have given thought to the backward-compatibility requirements of this change, and listed any mitigations above.
* [ ] (Optional) I have [run Routing Acceptance Tests and Routing Smoke Tests](https://github.com/cloudfoundry/routing-acceptance-tests/tree/e2a5b4eebc7e60615afb10ddbd250c5de73aa9fa#running-test-suites).
* [ ] (Optional) I have run [CF Acceptance Tests](https://github.com/cloudfoundry/cf-acceptance-tests#test-setup).

Commit:
Update README for Contribution

Add direnv to the list of required softwares

Closes: cloudfoundry/routing-release#352

peanball commented:
I would like to suggest adding the following under ["With Docker"](https://github.com/cloudfoundry/routing-release/pull/353/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R126):

```
> [!IMPORTANT]
> Make sure that [the prerequisites are installed](#developer-workflow) (`direnv`, checked out `wg-app-platform-runtime-ci` repo).
```

which should look as follows:

> [!IMPORTANT]
> Make sure that [the prerequisites are installed](#developer-workflow) (`direnv`, checked out `wg-app-platform-runtime-ci` repo).

(Those lines were not changed so I can't add a suggestion code fence...)

maxmoehl commented on README.md:
```diff
@@ -65,25 +65,20 @@ sticky sessions, refer to the [Session Affinity document](docs/session-affinity.
 ### <a name="developer-workflow"></a> Developer Workflow
 
-1. Clone CI repository (next to where routing-release is cloned)
+- Clone CI repository (next to where this code is cloned)
```
Can you please link the CI repository?

winkingturtle-vmw replied:
The link is present in the below snippet, but I've gone ahead and added it as a link too.

Commit:
Remove dependency on .envrc

- Reorganize scripts so that it's clear which scripts are ran
inside of the docker container and which are ran in the local workstation

- ./scripts/test-in-docker-locally.bash -> Helper to run everything
- ./scripts/create-docker-container.bash -> Create a docker containers to test and run things manually
